compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5001,6 +5008,45 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           val arg = genArgs1
           js.JSImportCall(arg)
 
+        case DYNAMIC_IMPORT =>
+          assert(args.size == 1,
+              s"Expected exactly 1 argument for JS primitive $code but got " +
+              s"${args.size} at $pos")
+
+          args.head match {
+            case Block(stats, Apply(fun @ Select(New(tpt), _), args)) =>
+              /* stats is always empty if no other compiler plugin is present.
+               * However, code instrumentation (notably scoverage) might add
+               * statements here. If this is the case, the thunk anonymous class
+               * has already been created when the other plugin runs (i.e. the
+               * plugin ran after jsinterop).
+               *
+               * Therefore, it is OK to leave the statements on our side of the
+               * dynamic loading boundary.
+               */
+
+              val clsSym = tpt.symbol
+              val ctor = fun.symbol
+
+              assert(clsSym.isSubClass(DynamicImportThunkClass),
+                  s"expected subclass of DynamicImportThunk, got: $clsSym")
+              assert(ctor.isPrimaryConstructor,
+                  s"expected primary constructor, got: $ctor")
```

Comment:
Perhaps include the current `pos` in the error messages, to ease debugging if this pops up in the wild.